torch/csrc/cuda/Module.cpp

```diff
@@ -26,6 +26,13 @@
 #include <c10/cuda/CUDAFunctions.h>
 #include <ATen/cuda/CUDAGraphsUtils.cuh>
 
+#if !defined(_MSC_VER)
+#include <sys/types.h>
+#include <unistd.h>
+#elif defined(_MSC_VER)
+#include <c10/util/win32-headers.h>
+#endif
```

Comment:
need header for getpid(). Follow this [example](https://github.com/pytorch/pytorch/blob/8d5f7ab06c0f2781d7195986a229162486c89fc0/aten/src/ATen/MapAllocator.cpp#L23-L28) for GetCurrentProcessId on Windows.

Reply:
Follow this [example](https://github.com/pytorch/pytorch/blob/main/c10/util/signal_handler.cpp#L167) to use syscall(SYS_gettid). This fixes some build issues since `gettid` is not available until [glibc 2.3.0](https://stackoverflow.com/questions/30680550/c-gettid-was-not-declared-in-this-scope).

Reply:
Could use portable std for this 
```
#include <thread>

auto tid = std::this_thread::get_id();
```